atlp-rwanda/Phantom-Backend-Elites
# Ft profile update

## What does this PR do?
- Added user profile update feature.

## Description of Tasks to be completed

- Added a function `updateProfile` to handle profile update in the `userController`
- Added a `route` which exactly points to that function
- Created a middleware `canUpdateProfile` to gather all the information needed from the user to check the following before a user updates his profile:
     - If the user is `logged in`. This means that a user should be having an `active token`
     - If user is an `operator` or `driver`
     - Note: Only a driver and operator are allowed to update their profiles. Other user including the `admin` are `prohibited`

## How should this be manually tested?
1. You should be logged in as an admin to create a new user. Use this route `http://localhost:5000/api/v1/users` to create a user

![image](https://user-images.githubusercontent.com/58140627/157245062-fc9d1209-936b-4ec2-9922-d8ed11883fdc.png)

2. Use credentials of the new registered user to login. Go to `http://localhost:5000/api/v1/auth/login`

![image](https://user-images.githubusercontent.com/58140627/157245454-c74e7f0b-84a6-4a09-98f6-357ddb19b10a.png)

3.  Copy the token above and set it in headers using `authorization` as a key to update your profile. Use this route `http://localhost:5000/api/v1/users/profile/update/` then add your `id`. Now ready to go! if nothing wrong, if you send the request your profile will be updated successfully. 

![image](https://user-images.githubusercontent.com/58140627/157248073-7f7f4842-e9a1-4713-873c-fc4f5192cc9e.png)

4. To verify what we have done,  let us try to display that user to see  if already updated. Use `http://localhost:5000/api/v1/users/` append the `id` of the user we recently created.
![image](https://user-images.githubusercontent.com/58140627/157250532-e9b648d4-ca86-46db-94f8-de0581c5fe00.png)

- [x] Thank you!
